src/Windowing/Extensions/Silk.NET.Windowing.Extensions.WebGPU/WebGPUWindow.cs

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Silk.NET.Core.Contexts;
+using Silk.NET.WebGPU;
+using Silk.NET.Windowing;
+
+namespace Silk.NET.Windowing.Extensions.WebGPU;
+/// <inheritdoc/>
+
+public static class WebGPUWindow
+{
+    /// <summary>
+    /// Creates a WebGPU surface from a window.
+    /// </summary>
+    /// <param name="wgpu">A WebGPU instance</param>
+    /// <param name="view">The window to get the surface from</param>
+    /// <returns>A newly created Surface*/</returns>
+    /// <exception cref="NotSupportedException">Throws when an unsupported platform is detected.</exception>
+    public static unsafe Surface* CreateWebGPUSurface(this IView view, NET.WebGPU.WebGPU wgpu, Instance* instance = null)
```

Comment:
Is it ever legal for the instance to be null? Surely we need something to make the surface on?

Reply:
It is legal, the instance is completely optional (in the triangle example we dont even create an instance, we just pass null)